mmv1/third_party/terraform/services/dataflow/resource_dataflow_flex_template_job.go.erb

```diff
@@ -275,35 +285,185 @@ func resourceDataflowFlexTemplateJobCreate(d *schema.ResourceData, meta interfac
 	return resourceDataflowFlexTemplateJobRead(d, meta)
 }
 
-func resourceDataflowFlexJobSetupEnv(d *schema.ResourceData, config *transport_tpg.Config) (dataflow.FlexTemplateRuntimeEnvironment, error) {
+func resourceDataflowFlexJobSetupEnv(d *schema.ResourceData, config *transport_tpg.Config) (dataflow.FlexTemplateRuntimeEnvironment, map[string]string, error) {
+
+	updatedParameters := tpgresource.ExpandStringMap(d, "parameters")
 
 	additionalExperiments := tpgresource.ConvertStringSet(d.Get("additional_experiments").(*schema.Set))
 
+	var autoscalingAlgorithm string
+	if p, ok := d.GetOk("parameters.autoscalingAlgorithm"); ok {
+		delete(updatedParameters, "autoscalingAlgorithm")
+		autoscalingAlgorithm = p.(string)
+	} else {
+		if v, ok := d.GetOk("autoscaling_algorithm"); ok {
+			autoscalingAlgorithm = v.(string)
+		}
+	}
+
+	var numWorkers int
+	if p, ok := d.GetOk("parameters.numWorkers"); ok {
+		number, err := strconv.Atoi(p.(string))
+		if err != nil {
+			return dataflow.FlexTemplateRuntimeEnvironment{}, updatedParameters, fmt.Errorf("parameters.numWorkers must have a valid integer assigned to it, current value is %s", p.(string))
+		}
+		delete(updatedParameters, "numWorkers")
+		numWorkers = number
+	} else {
+		if v, ok := d.GetOk("num_workers"); ok {
+			numWorkers = v.(int)
+		}
+	}
+
+	var maxNumWorkers int
+	if p, ok := d.GetOk("parameters.maxNumWorkers"); ok {
+		number, err := strconv.Atoi(p.(string))
+		if err != nil {
+			return dataflow.FlexTemplateRuntimeEnvironment{}, updatedParameters, fmt.Errorf("parameters.maxNumWorkers must have a valid integer assigned to it, current value is %s", p.(string))
+		}
+		delete(updatedParameters, "maxNumWorkers")
+		maxNumWorkers = number
+	} else {
+		if v, ok := d.GetOk("max_workers"); ok {
+			maxNumWorkers = v.(int)
+		}
+	}
+
+	var network string
+	if p, ok := d.GetOk("parameters.network"); ok {
+		delete(updatedParameters, "network")
+		network = p.(string)
+	} else {
+		if v, ok := d.GetOk("network"); ok {
+			network = v.(string)
+		}
+	}
+
+	var serviceAccountEmail string
+	if p, ok := d.GetOk("parameters.serviceAccountEmail"); ok {
+		delete(updatedParameters, "serviceAccountEmail")
+		serviceAccountEmail = p.(string)
+	} else {
+		if v, ok := d.GetOk("service_account_email"); ok {
+			serviceAccountEmail = v.(string)
+		}
+	}
+
+	var subnetwork string
+	if p, ok := d.GetOk("parameters.subnetwork"); ok {
+		d.Set("parameters.subnetwork", nil)
+		subnetwork = p.(string)
+	} else {
+		if v, ok := d.GetOk("subnetwork"); ok {
+			subnetwork = v.(string)
+		}
+	}
+
+	var tempLocation string
```

Comment:
Can we make the common examples of this block into a function?

Reply:
All moved to functions except the two int fields and the one bool.